content/en/blog/fec-with-pion.md

```diff
@@ -0,0 +1,145 @@
+   -> [RFC 8627, Section 1.1.4](https://datatracker.ietf.org/doc/rfc8627/#section-1.1.4)
+
+2. **Hybrid FEC and Retransmission Support**
+   The final spec explicitly supports both **proactive FEC** and **RTP retransmission**. It also requires that when FlexFEC is used alongside another retransmission mechanism, the answer in an SDP negotiation must use FlexFEC only, ensuring consistent protection handling.
+   -> [RFC 8627, Section 1.1.7](https://datatracker.ietf.org/doc/rfc8627/#section-1.1.7)
+
+3. **Improved Scalability and Consistency**
+   RFC 8627 addresses scalability limitations from earlier FEC mechanisms like [RFC 5109](https://datatracker.ietf.org/doc/html/rfc5109) and [RFC 2733](https://datatracker.ietf.org/doc/html/rfc2733). It also reuses the **RTP repair stream format** defined in [RFC 6363](https://datatracker.ietf.org/doc/html/rfc6363), making it more consistent with modern RTP practices and easier to interoperate with existing tooling.
+   -> [RFC 8627, Section 1.1.1](https://datatracker.ietf.org/doc/rfc8627/#section-1.1.1)
+
+4. **Packet and Header Format Clarifications**
+   The final spec clearly defines key header fields. For example, the **R bit** distinguishes retransmission (R=1) from FEC repair (R=0), while the **F bit** selects between flexible-mask mode (F=0) and fixed-offset mode (F=1). These explicit semantics simplify implementation logic and parsing.
+   -> [RFC 8627, Section 6.2](https://datatracker.ietf.org/doc/rfc8627/#section-6.2)
+
+### Enable FlexFEC Encoding in Pion
+
+Register FlexFEC codec to MediaEngine:
+
+```go
+err = m.RegisterCodec(
+    webrtc.RTPCodecParameters{
+        RTPCodecCapability: webrtc.RTPCodecCapability{
+            MimeType:     webrtc.MimeTypeFlexFEC03,
+            ClockRate:    90000,
+            Channels:     0,
+            SDPFmtpLine:  "repair-window=10000000",
+            RTCPFeedback: nil,
+        },
+        PayloadType: 118, // You can change this
+    },
+    webrtc.RTPCodecTypeVideo,
+)
+```
+
+Register FlexFEC interceptor:
+
+```go
+fecInterceptor, err := flexfec.NewFecInterceptor()
+// handle error...
+i.Add(fecInterceptor)
+```
+
+That's it! But with a few catches:
+1. The order of the FlexFEC interceptor in the chain of interceptors matters. It should be registered before TWCC header extension interceptor and NACK responder interceptor if you are using these.
+2. If you are using congestion controller, be aware of the bandwidth FEC is taking, you should set your encoder's bitrate to `cc_estimated_bandwidth - fec_bitrate`. By default, out of 5 media packets, 2 FEC packets will be produced, so you can calculate how much data encoder should output accordingly.
+
+### Configuring FlexFEC Encoder Interceptor
+
+There are [a few options](https://github.com/pion/interceptor/blob/master/pkg/flexfec/option.go) that you can use to tweak the encoder interceptor.
+
+### Verify that FlexFEC is Enabled
+
+- On sender side, which is within Pion, you can log the packets.
```

Comment:
It's not that easy to log something after the interceptors by hand, is it? Maybe we should suggest using the [packetdump](https://github.com/pion/interceptor/tree/master/pkg/packetdump) interceptor?